extensions/amp-timeago/0.2/timeago.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createElement, useRef, useState} from '../../../src/preact';
+import {timeago} from '../../../third_party/timeagojs/timeago';
+import {useIntersect} from './use-intersect';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+/**
+ * @param {!JsonObject} props
+ * @return {Preact.Renderable}
+ */
+export function Timeago(props) {
+  const {0: timestamp, 1: setTimestamp} = useState(
+    getFuzzyTimestampValue(
```

Comment:
This will have issues if, for instance, the `datetime` attribute on `<amp-timeago>` is mutated by `<amp-bind>`. Because this `useState` will only initialize once, we won't receive an updated value.

I think what we want is to remove this `useState`, and instead expose a way for `isIntersecting` to force a rerender (which will then force a rerender with an update-to-date fuzzy time).

Reply:
We had https://github.com/ampproject/amphtml/pull/25969/commits/a939b65b980f67d0b862fcad9eebd5ee3e14a045 originally, but didn't use it so we deleted it.

Reply:
as written, i think `useIntersect` will already cause a rerender as each intersection occurs and is added to the entries array

Reply:
I tested this out with amp-bind and `datetime` updates as expected. I added `useRerenderer` but am removing it now in favor of this original `useState` approach since it is more meaningful in terms of what the logic is intending to do.

Reply:
Can you upload your amp-bind test? I'm very confident this will not react correctly to changes only to attributes. (If `isInView` changes, it will correctly update)